cynic-parser/src/parser/schema.lalrpop

```diff
@@ -43,12 +43,25 @@ TypeSystemDefinition: DefinitionId = {
 }
 
 SchemaDefinition: SchemaDefinitionRecord = {
-    schema "{" <root_operations:RootOperationTypeDefinition*> "}" => SchemaDefinitionRecord {
+    schema <directives:Directives> <root_operations:RootOperationTypeDefinitions> => SchemaDefinitionRecord {
         description: None,
-        root_operations: ast.root_operation_definitions(root_operations),
+        directives,
+        root_operations,
     }
 };
 
+SchemaExtensionDefinition: SchemaDefinitionRecord = {
+    schema <directives:Directives> <root_operations:RootOperationTypeDefinitions?> => SchemaDefinitionRecord {
+        description: None,
+        directives,
+        root_operations: root_operations.unwrap_or_else(|| ast.no_root_operations()),
```

Comment:
I went with an empty range here, because making it optional would require either splitting SchemaDefinition and SchemaDefinitionExtension, or making it optional on SchemaDefinition.

Reply:
Yeah, empty range is all good - same as I've done for most other things

Reply:
There's an `impl Default for IdRange` defined here https://github.com/obmarg/cynic/blob/main/cynic-parser/src/type_system/ids.rs#L51-L54 so think this could just be:

```suggestion
        root_operations: root_operations.unwrap_or_default(),
```

Reply:
Nicer! Done.